app/src/main/java/com/asfoundation/wallet/redeem_gift/bottom_sheet/RedeemGiftBottomSheetFragment.kt

```diff
@@ -0,0 +1,207 @@
+      }
+      is Async.Loading -> {
+        if (clickAsync.value == null) {
+          showLoading()
+        }
+      }
+      is Async.Fail -> {
+        showErrorMessage(FailedRedeem.GenericError(""))
+      }
+      is Async.Success -> {
+        state.submitRedeemAsync.value?.let { redeemState ->
+          if (redeemState is SuccessfulRedeem)
+            showSuccess(redeemState)
+          else
+            showErrorMessage(redeemState as? FailedRedeem ?: FailedRedeem.GenericError(""))
+        }
+      }
+    }
+  }
+
+  override fun onSideEffect(sideEffect: RedeemGiftBottomSheetSideEffect) {
+    when (sideEffect) {
+      is RedeemGiftBottomSheetSideEffect.NavigateBack -> navigator.navigateBack()
+    }
+  }
+
+  private fun showErrorMessage(error: FailedRedeem) {
+    hideAll()
+    KeyboardUtils.hideKeyboard(view)
+    views.redeemGiftBottomSheetErrorAnimation.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetErrorAnimation.setImageResource(R.drawable.ic_alert_circle_red)
+    views.redeemGiftBottomSheetErrorAnimation.repeatCount = 0
+    views.redeemGiftBottomSheetErrorAnimation.playAnimation()
+    when(error) {
+      FailedRedeem.OnlyNewUsersError ->{
+        views.redeemGiftBottomSheetErrorTitle.text = getString(R.string.gift_card_error_new_users_title)
+        views.redeemGiftBottomSheetErrorSubtitle.text = getString(R.string.gift_card_error_try_with_different_body)
+      }
+      FailedRedeem.AlreadyRedeemedError ->{
+        views.redeemGiftBottomSheetErrorTitle.text = getString(R.string.gift_card_error_used_title)
+        views.redeemGiftBottomSheetErrorSubtitle.text = getString(R.string.gift_card_error_try_with_different_body)
+      }
+      is FailedRedeem.GenericError ->{
+        views.redeemGiftBottomSheetErrorTitle.text = getString(R.string.gift_card_error_general_title)
+        views.redeemGiftBottomSheetErrorSubtitle.text = getString(R.string.gift_card_error_general_body)
+      }
+    }
+    views.redeemGiftBottomSheetErrorTitle.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetErrorSubtitle.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetErrorButton.visibility = View.VISIBLE
+  }
+
+  private fun showLoading() {
+    hideAll()
+    views.redeemGiftBottomSheetSystemView.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetSystemView.showProgress(true)
+  }
+
+  private fun showDefaultScreen() {
+    hideAll()
+    views.redeemGiftBottomSheetString.setReadOnly(false, InputType.TYPE_CLASS_TEXT)
+    views.redeemGiftBottomSheetTitle.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetTextRectangle.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetTextRectangle.setBackgroundResource(
+      R.drawable.rectangle_text_redeem_gift)
+    views.redeemGiftBottomSheetSubmitButton.visibility = View.VISIBLE
+    views.redeemGiftBottomSheetSubmitButton.isEnabled = false
+  }
+
+  @SuppressLint("StringFormatMatches")
+  private fun showSuccess(redeemGift: RedeemCode) {
```

Comment:
We dont need this annotation since we are not passing any informtation to show on a string and the `redeemGift` argument is not being used